src/components/App.tsx

```diff
@@ -337,37 +338,79 @@ class App extends React.Component<any, AppState> {
     this.onSceneUpdated();
   };
 
+  private shouldForceLoadScene(
+    scene: ResolutionType<typeof loadScene>,
+  ): boolean {
+    if (!scene.elements.length) {
+      return true;
+    }
+
+    const roomMatch = getCollaborationLinkData(window.location.href);
+
+    if (!roomMatch) {
+      return false;
+    }
+    const collabForceLoadFlag = localStorage.getItem(
+      LOCAL_STORAGE_KEY_COLLAB_FORCE_FLAG,
+    );
+    if (collabForceLoadFlag) {
+      try {
+        const {
+          room: previousRoom,
+          timestamp,
+        }: { room: string; timestamp: number } = JSON.parse(
+          collabForceLoadFlag,
+        );
+        // if loading same room as the one previously unloaded within 15sec
+        //  force reload without prompting
```

Comment:
we save to local storage every 300ms so if user refreshes within less than 15 sec then by this logic it will replace the local storage right ?

Reply:
the localStorage item won't be overriden. The 15sec is just to ensure that if the user loads the url much later (which suggests it's not a browser refresh), it's not considered.